src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/InstantiateArgumentExceptionsCorrectlyTests.cs

```diff
@@ -593,6 +593,111 @@ End Sub
                End Class");
         }
 
+        [Fact]
+        public async Task ArgumentException_GenericParameterName_DoesNotWarn()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+                public class Class
+                {
+                    public void Test<TEnum>(TEnum first)
+                    {
+                        throw new System.ArgumentException(""first is incorrect"", nameof(TEnum));
+                    }
+                }");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+                Public Class [MyClass]
+                    Public Sub Test(Of TEnum)(ByVal first As TEnum)
+                        Throw New System.ArgumentException(""first is incorrect"", NameOf(TEnum))
+                    End Sub
+                End Class");
+        }
+
+
+        [Theory]
+        [InlineData("public", "dotnet_code_quality.api_surface = private", false)]
+        [InlineData("private", "dotnet_code_quality.api_surface = internal, public", false)]
+        [InlineData("public", "dotnet_code_quality.CA2208.api_surface = private, piblic", true)]
```

Comment:
Is "piblic" an intentional typo?

Reply:
huh, no, wondering how the test passed, maybe i did the typo after runnning tests, will check

Reply:
nothing wrong, public is default for SymbolVisibility enum, so anything non matching counted as public, anyways fixed typo,, thanks!